sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/VectorizedParquetRecordReader.java

```diff
@@ -265,7 +265,13 @@ private void initBatch(
       MemoryMode memMode,
       StructType partitionColumns,
       InternalRow partitionValues) {
-    StructType batchSchema = new StructType(sparkSchema.fields());
+    boolean returnNullStructIfAllFieldsMissing = configuration.getBoolean(
+      SQLConf$.MODULE$.LEGACY_PARQUET_RETURN_NULL_STRUCT_IF_ALL_FIELDS_MISSING().key(),
+      (boolean) SQLConf$.MODULE$.LEGACY_PARQUET_RETURN_NULL_STRUCT_IF_ALL_FIELDS_MISSING()
+        .defaultValue().get());
+    StructType batchSchema = !returnNullStructIfAllFieldsMissing
+      ? new StructType(sparkSchema.fields())
+      : (StructType) truncateType(sparkSchema, sparkRequestedSchema);
```

Comment:
the code before is `StructType batchSchema = new StructType(sparkSchema.fields());`, but now we only go to it if legacy config is off?

Reply:
ah I see, we need to call `truncateType` for the legacy code path, to deal with the extra cheap field we picked when all fields are missing. Can we add code comments to explain it?